src/main/scala/performanceanalysis/logreceiver/alert/AlertRuleActor.scala

```diff
@@ -19,15 +19,15 @@ class AlertRuleActor(alertingRule: AlertingRule, componentId: String, metricKey:
   lazy val actionActor: ActorRef = create(context)
 
   override def receive: Receive = {
-    case msg: CheckRuleBreak if doesBreakRule(msg.value) =>
+    case CheckRuleBreak(value) if value.isInstanceOf[Duration] && doesBreakRule(value.asInstanceOf[Duration]) =>
       log.info(s"Rule $alertingRule is broken for $componentId/$metricKey")
       actionActor ! Action(alertingRule.action.url,
       s"Rule $alertingRule was broken for component id $componentId and metric key $metricKey")
```

Comment:
IMHO For readability it is better to put message in `val`. It is trivial, if you dont agree ignore me